WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/AddProductToOrderViewModel.swift

```diff
@@ -178,7 +184,11 @@ extension AddProductToOrderViewModel: SyncingCoordinatorDelegate {
                 DDLogError("⛔️ Error searching products during order creation: \(error)")
             }
 
-            self.transitionToResultsUpdatedState()
+            // Don't update sync status if this isn't the latest search term.
+            // Avoids a flash of the empty results view if a new search is being performed.
+            if keyword == self.searchTerm {
+                self.transitionToResultsUpdatedState()
+            }
```

Comment:
I see, shouldn't we totally avoid any kind of logic if we get a result for a different search term?
I'm assuming this happens because one network request came slower than the next one.

I guess this could be handled if we were able to cancel the current request before firing the new one, right? 🤔

Reply:
> I see, shouldn't we totally avoid any kind of logic if we get a result for a different search term?

Hm, good point. I was thinking that the only thing we needed to avoid was updating the sync status, but it would be better to avoid any other logic. This seems especially important since other logic could potentially affect the UI in a confusing way, e.g. the error notices we'll be adding, so I went ahead and updated it in [2455207](https://github.com/woocommerce/woocommerce-ios/pull/6112/commits/24552075f140bdb59b9edc29c5d0a60845ddddbb).

> I guess this could be handled if we were able to cancel the current request before firing the new one, right?

Yeah, ideally we could cancel the previous search request if we're sending another one. My understanding is we can't do that with the way we're dispatching these actions. We could also mitigate this by increasing the debounce time before sending a search request, but that wouldn't entirely prevent this.